catalyst/core/runner.py

```diff
@@ -367,7 +368,12 @@ def _run_batch(self, batch: Mapping[str, Any]) -> None:
             batch (Mapping[str, Any]): dictionary with data batches
                 from DataLoader.
         """
-        self.state.global_step += self.state.batch_size
+        if isinstance(batch, list):
+            self.state.batch_size = len(batch[0])
+        else:
+            self.state.batch_size = next(iter(batch.values())).shape[0]
```

Comment:
moreover, what do you think about performance of this solution?

Reply:
python `if` statement is really computation heavy, but same is deep learning pipelines :)

Reply:
I hate this solution at all.
It would be better if there were only dict-like batches in catalyst, IMHO.
_(or even better, if torch dataloaders had something like `next_batch_size` property, but I definitely cannot do anything with it)_

This does not seem to affect the performance very much. but ok, will do some formal tests.